### ISSUE TITLE
Auto-optimization breaks cmd+option+v for moving files in Finder

### ISSUE THREAD

**Issue by diogox** [FuzzyIdeas/Clop#8]
In Finder, when you want to move files with the keyboard, you have to use cmd+c and then cmd+option+v. There's no ctrl+x and ctrl+v workflow like in Linux and Windows.

With Clop installed, when you cmd+c an image, it creates a new optimized version. And when you cmd+option+v, it moves the optimized version to the desired location, leaving the original in place.

While I'm glad the original is safe, this breaks the keyboard-driven workflow for moving files.

I read [this issue](https://github.com/FuzzyIdeas/Clop/issues/6#issuecomment-1685406745) where you mention we could have the optimization happen on paste with cmd+v, but that it needs more permissions.

Personally, I'm fine with needing more permissions and would love to have this option 🙌

**Comment by alin23** [comment]
In the next version (to be released) I made an important change to help with this issue:

When copying more than one file, the optimisation does not happen. 

I think that would fix your use case without needing additional permissions, and a lot of complexity on my part. 

Would you like to try a beta?

**Comment by diogox** [comment]
I appreciate your answer :)

That sounds like an improvement. While I’d be happy to give it a try, moving one file is still a use case for me.

I prefer my apps to be “set it and forget it”, and this would be an edge case that I’d have to keep in mind.

Tbh my ideal use case would be to have flameshot screenshots automatically optimize when I copy them to the clipboard, and nothing else. But I don’t see a way of doing just that 🤔 

Perhaps a hotkey to “Enable clipboard optimization for next one”? But that’s a different request altogether 😄

**Comment by alin23** [comment]
Curious if you already know about the **Pause** hotkey in Clop (<kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>P</kbd> by default). It's the opposite of what you asked for, you press it before copying a file, and Clop ignores that file.

**Comment by diogox** [comment]
I’ve seen it, but like I mentioned, I don’t want to have to remember hotkeys for specific tasks.

Even with the opposite hotkey, for enabling optimization for the next clipboard, I would just write a Hammerspoon automation so I wouldn’t have to think about it.

It’s not really feasible to do it with the Pause option :/

**Comment by alin23** [comment]
So your ideal case would be:

- <kbd>Cmd</kbd>-<kbd>C</kbd>: copy and don't trigger Clop
- <kbd>Cmd</kbd>-<kbd>Opt</kbd>-<kbd>V</kbd>: move what was copied as normal
- <kbd>Cmd</kbd>-<kbd>V</kbd>: paste optimised image

There would be no way to paste a non-optimised image.

Am I understanding this correctly? I'd like to try an implementation but I want to be sure I have the requirements right.

**Comment by diogox** [comment]
Personally, I would like to only optimize images that are going to leave my computer (when I'm sending them over the network, so that they get sent faster).

I don't know if this would be everyone's use case.

So, at least for me, what you described seems mostly accurate. But if you had a way of telling when `Cmd`-`V` is being used locally (in Finder), and not trigger an optimization, that would be even better. As it would allow me to keep the original copy/paste/move functionality, while making image transfers online (mostly screenshots to illustrate a point) a lot more efficient.

Does that make sense?

As I wrote this, I actually remembered another option: If clop could watch a directory (where I can set my screenshot tool to send images automatically), and automatically optimize any new images and add them to the clipboard. That might be a simpler approach. For my use case at least 🤔

**Comment by alin23** [comment]
It's quite complex to identify a local vs network paste. I'll look into it. 

The folder watching part is already implemented, check out Clop's settings under the Images tab. 

The watched optimized files are not added to clipboard automatically yet, I have to implement that, but you can at least drag the thumbnail result.

**Comment by diogox** [comment]
Thanks for considering this feature, I appreciate it :)

**Comment by alin23** [comment]
I decided to not add Accessibility Permissions (for now). Mostly because of time constraints, it took a lot of time to do the priority features and fixes for 2.2.0 and I have to take a break.

I added it to my todo list. Until then, 2.2.0 fixes part of this problem by:
- ignoring <kbd>Cmd-C</kbd> of multiple files
- having an option to add optimised files from watched dirs to clipboard
- providing the global <kbd>Ctrl-Shift-P</kbd> hotkey for pausing Clop for the next <kbd>Cmd-C</kbd>
